src/Dfe.PlanTech.Application/Responses/Commands/ProcessCheckAnswerDtoCommand.cs

```diff
@@ -33,27 +33,8 @@ private static CheckAnswerDto RemoveDetachedQuestions(CheckAnswerDto checkAnswer
         var questionWithAnswerMap = checkAnswerDto.Responses.ToDictionary(questionWithAnswer => questionWithAnswer.QuestionRef,
```

Comment:
Unsure to be honest!

I'm assuming this was added originally for optimal lookups, however without benchmarking it I'm unsure of the exact benefit of it vs the cost of extra allocation and initial lookup to create the dictionary.

I've left it as is for now as it doesn't seem to be impacting anything, and is working, but it would be good to benchmark in the future.